src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PlacementApplicationEntity.kt

```diff
@@ -150,18 +150,22 @@ enum class PlacementType {
   ADDITIONAL_PLACEMENT,
 }
 
-enum class PlacementApplicationDecision {
-  ACCEPTED,
-  REJECTED,
-  WITHDRAW,
-  WITHDRAWN_BY_PP,
+enum class PlacementApplicationDecision(val apiValue: ApiPlacementApplicationDecision) {
+  ACCEPTED(ApiPlacementApplicationDecision.accepted),
+  REJECTED(ApiPlacementApplicationDecision.rejected),
+  WITHDRAW(ApiPlacementApplicationDecision.withdraw),
+  WITHDRAWN_BY_PP(ApiPlacementApplicationDecision.withdrawnByPp),
   ;
 
-  fun convertToApi() = when (this) {
-    ACCEPTED -> ApiPlacementApplicationDecision.accepted
-    REJECTED -> ApiPlacementApplicationDecision.rejected
-    WITHDRAW -> ApiPlacementApplicationDecision.withdraw
-    WITHDRAWN_BY_PP -> ApiPlacementApplicationDecision.withdrawnByPp
+  companion object {
+    fun valueOf(apiValue: ApiPlacementApplicationDecision): PlacementApplicationDecision? {
+      for (e in entries) {
+        if (e.apiValue == apiValue) {
+          return e
+        }
+      }
+      return null
```

Comment:
i was playing around with interfaces, enums and extension functions to see if this could be done in a generic way and i couldn't get it to work (with extension functions at least). But maybe we could at least delegate to another functions for the for loop bit. i'll give it a go next time we add an enum

Reply:
Yeah, that would be good. It's definitely a pattern that will be useful in a ton of places!